File: 2022/2022W01/preppindataW01.R
library(dplyr)
library(readr)
library(lubridate)

final <- read_csv("C:/Data/PreppinData/PD 2022 Wk 1 Input - Input.csv", 
                  col_types = cols(`Date of Birth` = col_date(format = "%m/%d/%Y"))) %>%
  mutate('Pupil\'s Name' = paste(`pupil last name`, `pupil first name`, sep=', '),
         'parent first name' = if_else(`Parental Contact`==1, `Parental Contact Name_1`, `Parental Contact Name_2`),
         'Parental Contact Full Name' = paste(`pupil last name`, `parent first name`, sep=', '),
         'Parental Contact Email Address' =  paste0(`parent first name`, '.', `pupil last name`, '@', `Preferred Contact Employer`, '.com'),
         'Birth Fiscal Year' = as.integer(quarter(`Date of Birth`,with_year = TRUE, fiscal_start = 9)),
         'Academic Year' = 2016 - if_else(`Birth Fiscal Year`>2015, as.integer(2015), `Birth Fiscal Year`)) %>%
  select(c('Academic Year', 'Pupil\'s Name', 'Parental Contact Full Name', 'Parental Contact Email Address'))
  
View(final)

File: 2022/2022W01/PreppindataW01.py
import pandas as pd
import fiscalyear

fiscalyear.START_MONTH = 9
final = pd.read_csv(r'C:/Data/PreppinData/PD 2022 Wk 1 Input - Input.csv',
                    parse_dates=['Date of Birth'], date_parser=lambda x: pd.to_datetime(x, format='%m/%d/%Y'))
final['Pupil\'s Name'] = final['pupil last name']+', '+final['pupil first name']
final['parent first name'] = final.apply(lambda x: x['Parental Contact Name_1'] if x['Parental Contact']==1 else x['Parental Contact Name_2'], axis=1)
final['Parental Contact Full Name'] = final['pupil last name']+', '+final['parent first name']
final['Parental Contact Email Address'] = final['parent first name']+'.'+final['pupil last name']+'@'+final['Preferred Contact Employer']+'.com'
final['Academic Year'] = final['Date of Birth'].apply(lambda x: 2016 - min(fiscalyear.FiscalDate(x.year, x.month, x.day).fiscal_year,2015))
final = final[['Academic Year', 'Pupil\'s Name', 'Parental Contact Full Name', 'Parental Contact Email Address']]
